src/models/reach_curve.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for modeling a reach curve.
+
+A reach curve is a mapping from scalar spend or impression value to reach.
+"""
+
+import copy
+from typing import Callable
+
+from wfa_planning_evaluation_framework.models.reach_point import ReachPoint
+from wfa_planning_evaluation_framework.models.reach_surface import ReachSurface
+
+
+class ReachCurve(ReachSurface):
+  """Models reach as a function of scalar spend or impressions."""
+
+  def __init__(self, data: [ReachPoint], max_reach: int = None):
+    """Constructor
+
+    Args:
+      data:  A list of ReachPoints to which the model is to be fit.
+      max_reach:  Optional.  If specified, the maximum possible reach that can
+        be achieved.
+    """
+    super().__init__(data, max_reach)
+    if len(data[0].impressions) != 1:
+      raise ValueError(
+          "Attempt to create reach curve from multidimensional data")
+
+  def _by_quantile(self, quantile: float,
+                   reach_function: Callable[[int], ReachPoint]) -> int:
+    """Returns a value needed to achieve a given quantile of 1+ reach.
+
+    Args:
+      quantile: float, a number between 0 and 1, excluding the endpoints.
```

Comment:
s/between 0 and 1/in (0, 1) ? also in the next function

Reply:
Done.